### PR TITLE
feat(acl): add support for configuring mobile menu permissions

### PR DESCRIPTION
**Pull request by zhangzhonghe**
### This is a ...
- [x] New feature
- [ ] Improvement
- [ ] Bug fix
- [ ] Others

### Motivation

Like desktop, mobile allows you to configure a role's menu access.
### Description 

1. Refactor the "Users &amp; Permissions" configuration page (in the location shown below) to make it extensible.
image

2. Add the "Mobile menu" configuration item to configure the permissions of the mobile menu.
3. New `mobileRoutes:listAccessible` interface (returns only menus accessible to the current role) to distinguish it from `mobileRoutes:list` (which returns all menus and is only accessible to roles with administrator rights).

### Showcase

https://github.com/user-attachments/assets/93a9e37d-edce-4f55-8fd9-6079b8cdbc14
### Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      Support for configuring permissions for mobile menus.     |
| 🇨🇳 Chinese |     支持配置移动端菜单的权限。      |

### Docs

| Language   | Link |
| ---------- | --------- |
| 🇺🇸 English |  [Access Control](https://docs.nocobase.com/handbook/acl)    |
| 🇨🇳 Chinese |  [权限控制](https://docs-cn.nocobase.com/handbook/acl)  |

### Checklists
- [x] All changes have been self-tested and work as expected
- [x] Test cases are updated/provided or not needed
- [x] Doc is updated/provided or not needed
- [x] Component demo is updated/provided or not needed
- [x] Changelog is provided or not needed
- [x] Request a code review if it is necessary

### CODE REVIEW

File: packages/plugins/@nocobase/plugin-mobile/src/server/collections/extendRoleField.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ */
+
+import { extendCollection } from '@nocobase/database';
+
+export default extendCollection({
+  name: 'roles',
+  fields: [
+    {
+      type: 'belongsToMany',
+      name: 'mobileMenuUiSchemas',
```
Comment (chenos) [inline packages/plugins/@nocobase/plugin-mobile/src/server/collections/extendRoleField.ts:17]: 这个 name 叫 `mobileRoutes` 比较合适，是 roles 和 mobileRoutes 两个表之间的关系，跟 UISchema 没啥关系

File: packages/plugins/@nocobase/plugin-mobile/src/server/plugin.ts [hunk 2]
```diff
@@ -21,7 +56,97 @@ export class PluginMobileServer extends Plugin {
...
+    this.app.acl.registerSnippet({
+      name: `pm.${this.name}.roles`,
+      actions: ['roles.mobileMenuUiSchemas:*'],
+    });
+  }
+
+  /**
+   * used to implement: roles with permission (allowNewMobileMenu is true) can directly access the newly created menu
+   */
+  bindNewMenuToRoles() {
```
Comment (chenos) [inline packages/plugins/@nocobase/plugin-mobile/src/server/plugin.ts:70]: 这个用 db.on 的事件比较合适
```ts
// 创建之后，需要事务一致性 transaction
db.on('mobileRoutes:afterCreate', async (instance,  { transaction }), {
  db.getRepository();
});
```

File: packages/plugins/@nocobase/plugin-mobile/src/server/plugin.ts [hunk 3]
```diff
@@ -21,7 +56,97 @@ export class PluginMobileServer extends Plugin {
       actions: ['mobileRoutes:list'],
     });
 
-    this.app.acl.allow('mobileRoutes', 'list', 'loggedIn');
+    this.app.acl.allow('mobileRoutes', 'listAccessible', 'loggedIn');
+    this.app.acl.allow('uiSchemas', 'getMobileJsonSchema', 'loggedIn');
```
Comment (chenos) [inline packages/plugins/@nocobase/plugin-mobile/src/server/plugin.ts:60]: 没有 getMobileJsonSchema 这个 api 了

File: packages/plugins/@nocobase/plugin-mobile/src/server/plugin.ts [hunk 4]
```diff
@@ -21,7 +56,97 @@ export class PluginMobileServer extends Plugin {
...
+  //     ctx.body = {};
+  //     await next();
+  //   });
+  // }
+
+  registerActionHandlers() {
+    this.app.resourceManager.registerActionHandler('mobileRoutes:listAccessible', async (ctx, next) => {
+      const mobileRoutesRepository = ctx.db.getRepository('mobileRoutes');
+      const rolesRepository = ctx.db.getRepository('roles');
+      const items = await mobileRoutesRepository.find({
```
Comment (chenos) [inline packages/plugins/@nocobase/plugin-mobile/src/server/plugin.ts:128]: 这里用树表查询感觉比较方便：

```ts
find({ tree: true, filter: { id: ids } });
```

File: packages/plugins/@nocobase/plugin-mobile/src/server/plugin.ts [hunk 5]
```diff
@@ -21,7 +56,97 @@ export class PluginMobileServer extends Plugin {
...
+        return await next();
+      }
+
+      const role = await rolesRepository.findOne({
+        filterByTk: ctx.state.currentRole,
+        appends: ['mobileMenuUiSchemas'],
+      });
+
+      const mobileMenuUiSchemas = role.get('mobileMenuUiSchemas').map((item) => item.id);
+      const result = filterMobileRoutes(items, (item) => mobileMenuUiSchemas.includes(item.id));
```
Comment (chenos) [inline packages/plugins/@nocobase/plugin-mobile/src/server/plugin.ts:144]: 直接用树表查询 + filter 比较方便

File: packages/plugins/@nocobase/plugin-mobile/src/server/collections/extendRoleField.ts (hunk 1 above)
Comment (chenos) [reply packages/plugins/@nocobase/plugin-mobile/src/server/collections/extendRoleField.ts:17]: 要处理删除逻辑，可以加个 onDelete 的参数配置

```
onDelete: 'CASCADE',
```

File: packages/plugins/@nocobase/plugin-mobile/src/server/plugin.ts [hunk 6]
```diff
@@ -21,7 +56,97 @@ export class PluginMobileServer extends Plugin {
       actions: ['mobileRoutes:list'],
     });
 
-    this.app.acl.allow('mobileRoutes', 'list', 'loggedIn');
+    this.app.acl.allow('mobileRoutes', 'listAccessible', 'loggedIn');
+    this.app.acl.allow('uiSchemas', 'getMobileJsonSchema', 'loggedIn');
+    this.app.acl.registerSnippet({
+      name: `pm.${this.name}.roles`,
```
Comment (chenos) [inline packages/plugins/@nocobase/plugin-mobile/src/server/plugin.ts:62]: 这里的 Snippet 得用 `pm.mobile`

File: packages/plugins/@nocobase/plugin-mobile/src/server/plugin.ts (hunk 5 above)
Comment (zhangzhonghe) [reply packages/plugins/@nocobase/plugin-mobile/src/server/plugin.ts:144]: 这里树表的 filter 要怎么写？

**Pull request merged**